acoustic_msgs/README.md

```diff
@@ -44,14 +44,13 @@ adheres to these guidelines:
     * Publish a separate message with the additional data, but no duplication
       of the standard message, following the Image/CameraInfo pattern.
   - In order to support generality for a wide range of sensors, there will be some
-    fields that are not applicable to some sensors.
+    fields that are not applicable to some sensors. The length of the array should
+    match the length of the data, even if every value is identical.
     (e.g. per-beam elevation angles; per-beam beamwidth)
-    * If not supported at all, leave that array empty
```

Comment:
Do we have an example of this? (And that our existing tools support?)

I think we should ditch the length-1 thing, since it makes parsing them messier and I don't think any of the code I'm familiar with uses it.

Reply:
Pinging @CaptKrasno and @rolker about this convention.

I'd like to revisit allowing length-one arrays since we're about to release these messages. I'm concerned because I don't think current code uniformly supports this, and we'd be committing ourselves to more complicated code for what is a minor (and often not applicable) space savings in the messages.

Reply:
I think RawSonarImage's float32[] tx_delays is an example where knowing that all the values are the same can lead to simpler (and more efficient) code for processing or visualization. A constant or 0 tx delay means the beams don't need to be individually shifted and the whole ping's data array can be directly loaded in texture memory for visualization for example.
Looking through my code, I set the tx_delay  to 0 for the single beam EK80 but in the rviz plugin, I don't even look at it for the moment. (The rviz plugin is still a prototype, tx_delays is one thing that will need to be handled.)
Thinking ahead to the Konsberg EM2040 that I'll probably tackle eventually, I suspect that data volumes are going to make potential simpler processing more significant.
Of course, code could run through the array before preocessing the data to determine if the values are all the same then take the simple path if so, so I'm not rulling out always having full arrays.

Reply:
I think I could really go either way on this.  I guess we need to decide if our priority is parseability or computation efficiency when we have a conflict like this.

It's worth considering that you will already need to check if the array is length zero.  Is it a big deal to check if it is length 1 too?   I'm not sure.

Reply:
> It's worth considering that you will already need to check if the array is length zero. Is it a big deal to check if it is length 1 too? I'm not sure.

In the text of this suggested change, I was also proposing that we don't allow length-zero arrays either, instead expecting the data provider to set every field. If there IS a default value applied on the consumer side, we need to document it  (and don't currently).

Reply:
My reasoning for this was:
* explicit is better than implicit => don't allow empty arrays
* messages are write-once read-many-times, so make the code for parsing them as simple as possible to get right.

So, on the "parseability or computation efficiency" question that @CaptKrasno brings up, I care more about parsing.

Reply:
It seems nobody feels super strongly about this, and I see both sides, so I'll revert this suggested change and let the previous convention stand.